### PR TITLE
Windows 64.

### PR DESCRIPTION
**Pull request by kntroh**
Fixes for Windows 64-bit.

I tested all snippets and my application on Windows 7 64-bit with VS 2013 and Windows SDK for 8.1.

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -87,6 +91,14 @@ For Windows:
     -L+org.eclipse.swt.win32.win32.x86.lib -L+dwt-base.lib -L/SUBSYSTEM:WINDOWS:4.0
   ```
 
+  If you use Microsoft linker:
+
+  ```
+  $ dmd main.d -I<dwt>\imp -J<dwt>\res -L/LIBPATH:<dwt>\lib ^
```
Comment (jacob-carlborg) [inline README.markdown]: The -J flag doesn't match the one for DMC linker.

File: build.d [hunk 2]
```diff
@@ -321,13 +324,26 @@ void buildApp( string basedir, string srcdir, string resdir, in string[] dflags,
     }
 
     static if (isWindows) {
-        rsp ~= "-L/NOM";
-        rsp ~= "-L/SUBSYSTEM:CONSOLE:4.0";
-        rsp ~= win_path(.absolutePath("win-res\\resource.res"));
-        foreach (libname; libnames) {
-            rsp ~= "-L+" ~ libname ~ LIBEXT;
+        if (extraOptions.countUntil("-m64") == -1) {
```
Comment (jacob-carlborg) [inline build.d]: I think `canFind` is more appropriate.

**Comment by jacob-carlborg** [comment]
Would it be difficult to extend this build script to support 32bit COFF?

File: README.markdown (hunk 1 above)
Comment (kntroh) [inline README.markdown]: `-J<dwt>\res` is required for compiler. dmd reads `res\org.eclipse.swt.internal.SWTMessages_*.properties` when compile time.

File: build.d (hunk 2 above)
Comment (kntroh) [inline build.d]: I'll replace it.

**Comment by kntroh** [comment]
> Would it be difficult to extend this build script to support 32bit COFF? 

I think that it is possible, but need a new option. (Windows dmd user uses OPTLINK normally.)

**Comment by kntroh** [comment]
I updated pull request.

File: README.markdown (hunk 1 above)
Comment (jacob-carlborg) [inline README.markdown]: Yes, but the existing instructions says `-J<dwt>\org.eclipse.swt.win32.win32.x86\res`.

**Comment by jacob-carlborg** [comment]
> I think that it is possible, but need a new option. (Windows dmd user uses OPTLINK normally.)

I think we can use the same flag as DMD uses, `-m32mscoff`.

File: README.markdown (hunk 1 above)
Comment (kntroh) [inline README.markdown]: The contents of the folder is copied to `<dwt>\res` when building.

Examples of 32-bit and 64-bit seems to right both, What had better fix either?

**Comment by etcimon** [comment]
I actually have a dub.json file on my fork that supports 32 bit ms coff and integrates nicely to dub projects. It needs removal of the pragma instructions in the code and doesn't have resource compiler configured yet but I really needed it for my projects if interested: https://github.com/etcimon/dwt/blob/master/dub.json

**Comment by etcimon** [comment]
How do you avoid the errors related to int-long implicit conversion? I tried to compile with -m64 but couldn't due to these

**Comment by kntroh** [comment]
I am trying to support for ms32coff now. but more difficult than I thought, take time.

> How do you avoid the errors related to int-long implicit conversion? I tried to compile with -m64 but couldn't due to these

It is brute force. I fixed all a signature of API functions, and removed compile errors. then checked all cast, and tested all snippets and DWT applications, and I fixed occured problem.

**Comment by etcimon** [comment]
> It is brute force. I fixed all a signature of API functions, and removed compile errors. then checked all cast, and tested all snippets and DWT applications, and I fixed occured problem.

I did that too it took hours but then winapi failed at createHandle. Can you share a link to this?

**Comment by etcimon** [comment]
Ok i found the other pull. Thanks for the nice work

**Comment by etcimon** [comment]
> I am trying to support for ms32coff now. but more difficult than I thought, take time.

You need to edit win64.mak. I changed it to MODEL=32mscoff and remove all occurence of amd64/ in the file (there are 3), for both druntime and phobos. Save this to win32mscoff.mak 

You need to place the phobos32mscoff.lib into dmd2/windows/lib32mscoff/ (the folder doesn't exist).

**Comment by kntroh** [comment]
I updated this pull request for support for 32bit COFF now.

Thanks @etcimon.

File: README.markdown (hunk 1 above)
Comment (jacob-carlborg) [inline README.markdown]: I don't know which is best, but I should be consistent.

Comment (kntroh) [inline README.markdown]: I think `-J<dwt>\res` is proper maybe.

I'll modify an example for optlink.

Comment (kntroh) [inline README.markdown]: I updated it.

**Comment by jacob-carlborg** [comment]
@kntroh could you please mention the `-m32mscoff` flag in the documentation.

**Comment by kntroh** [comment]
Added it.

**Comment by jacob-carlborg** [comment]
@kntroh Awesome work as always, thanks.

**Pull request merged**